app/models/manageiq/providers/ansible_tower/automation_manager/refresher.rb

```diff
@@ -15,18 +16,23 @@ def collect_inventory_for_targets(ems, targets)
 
           _log.info "Collecting inventory for #{target.class} [#{target_name}] id: [#{target.id}]..."
 
-          inventory = ManageIQ::Providers::AnsibleTower::AutomationManager::RefreshWorker::Collector.new(ems)
+          # actually the collector could be build in parse_targeted_inventory
+          collector = ManageIQ::Providers::AnsibleTower::Inventory::Collectors::AutomationManager.new(ems)
 
           _log.info "Collecting inventory...Complete"
-          [target, inventory]
+          [target, collector]
         end
       end
 
-      def parse_targeted_inventory(ems, target, inventory)
+      # FIXME: this can be done in BaseManager
+      def parse_targeted_inventory(ems, target, collector)
         log_header = format_ems_for_logging(ems)
         _log.debug "#{log_header} Parsing inventory..."
         hashes, = Benchmark.realtime_block(:parse_inventory) do
-          ManageIQ::Providers::AnsibleTower::AutomationManager::RefreshWorker::Parser.new(ems, target, inventory).parse
+          inv_target = ManageIQ::Providers::AnsibleTower::Inventory::Targets::AutomationManager.new(target)
```

Comment:
Wrapping my head around this ... when the target is the manager, then it's a full refresh?